src/workerd/api/container.h

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2025 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#pragma once
+// APIs that an Actor (Durable Object) uses to access its own state.
+//
+// See actor.h for APIs used by other Workers to talk to Actors.
+
+#include <workerd/io/container.capnp.h>
+#include <workerd/io/io-own.h>
+#include <workerd/jsg/jsg.h>
+
+namespace workerd::api {
+
+class Fetcher;
+
+// Implements the `ctx.container` API for durable-object-attached containers. This API allows
+// the DO to supervise the attached container (lightweight virtual machine), including starting,
+// stopping, monitoring, making requests to the container, intercepting outgoing network requests,
+// etc.
+class Container: public jsg::Object {
+ public:
+  Container(rpc::Container::Client rpcClient);
+
+  struct StartupOptions {
+    jsg::Optional<kj::Array<kj::String>> entrypoint;
+    bool enableInternet = false;
+
+    // TODO(containers): stdio interception
```

Comment:
Nit: would be helpful if you could expand this comment to provide more detail about what is intended for someone else coming into this code later on.